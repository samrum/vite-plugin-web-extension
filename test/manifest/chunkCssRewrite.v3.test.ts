import { getResourceDir, runManifestV3Test } from "./manifestTestUtils";

const resourceDir = getResourceDir("chunkCssRewrite");

runManifestV3Test("chunkCssRewrite", () => ({
  content_scripts: [
    {
      js: [`${resourceDir}/content1.js`],
      matches: ["https://*/*", "http://*/*"],
    },
    {
      js: [`${resourceDir}/content2.js`],
      matches: ["https://*/*", "http://*/*"],
    },
    {
      js: [`${resourceDir}/contentNoCss.js`],
      matches: ["https://*/*", "http://*/*"],
    },
  ],
}));
