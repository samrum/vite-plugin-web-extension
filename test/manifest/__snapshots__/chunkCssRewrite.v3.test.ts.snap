// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`chunkCssRewrite - Manifest V3 1`] = `
[
  {
    "fileName": "assets/content1.css",
    "name": "content1.css",
    "source": "#content1 {
  display: flex;
}
",
    "type": "asset",
  },
  {
    "fileName": "assets/content2.css",
    "name": "content2.css",
    "source": "#content2 {
  display: flex;
}
",
    "type": "asset",
  },
  {
    "fileName": "assets/contentShared.css",
    "name": "contentShared.css",
    "source": "#contentShared {
  display: flex;
}
",
    "type": "asset",
  },
  {
    "fileName": "manifest.json",
    "name": undefined,
    "source": "{
  "version": "1.0.0",
  "name": "Manifest Name",
  "manifest_version": 3,
  "content_scripts": [
    {
      "js": [
        "test/manifest/resources/chunkCssRewrite/content1.js"
      ],
      "matches": [
        "https://*/*",
        "http://*/*"
      ]
    },
    {
      "js": [
        "test/manifest/resources/chunkCssRewrite/content2.js"
      ],
      "matches": [
        "https://*/*",
        "http://*/*"
      ]
    },
    {
      "js": [
        "test/manifest/resources/chunkCssRewrite/contentNoCss.js"
      ],
      "matches": [
        "https://*/*",
        "http://*/*"
      ]
    }
  ],
  "web_accessible_resources": [
    {
      "resources": [
        "assets/content1.css",
        "assets/content2.css",
        "assets/contentShared.css"
      ],
      "matches": [
        "https://*/*",
        "http://*/*"
      ],
      "use_dynamic_url": true
    }
  ]
}",
    "type": "asset",
  },
  {
    "fileName": "test/manifest/resources/chunkCssRewrite/content1.js",
    "name": undefined,
    "source": "/* empty css                        */
console.log(["assets/content1.css","assets/contentShared.css"]);
",
    "type": "asset",
  },
  {
    "fileName": "test/manifest/resources/chunkCssRewrite/content2.js",
    "name": undefined,
    "source": "/* empty css                        */
console.log(["assets/content2.css","assets/contentShared.css"]);
",
    "type": "asset",
  },
  {
    "fileName": "test/manifest/resources/chunkCssRewrite/contentNoCss.js",
    "name": undefined,
    "source": "console.log([]);
",
    "type": "asset",
  },
]
`;
